### PR TITLE
nixos/nncp: add caller and daemon services

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

We have `programs.nncp` but no `services.nncp`.

## Things done

- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/nncp.nix [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      };
+
+      daemon = {
+        enable = mkEnableOption ''
+          NNCP TCP synronization daemon.
+          The daemon will take configuration from
+          [](#opt-programs.nncp.settings)
+        '';
+        socketActivation = {
+          enable = mkEnableOption ''
```
Comment (Mic92) [inline nixos/modules/services/networking/nncp.nix:37]: Sounds like a useful thing to have enabled by default, no?

Comment (Mic92) [reply nixos/modules/services/networking/nncp.nix:37]: could be done later, I guess. Works otherwise.

Comment (ehmry) [reply nixos/modules/services/networking/nncp.nix:37]: The socket activation is designed to work with [UCSPI-TCP](http://cr.yp.to/ucspi-tcp.html) and not systemd, so I don't use it and haven't tested it well. I also run the daemon persistently for the sake of periodic multicast announcements.

Comment (Mic92) [reply nixos/modules/services/networking/nncp.nix:37]: I tested it.

**Pull request merged**